wd_helper.au3

```diff
@@ -1187,6 +1187,104 @@ Func _WD_LoadWait($sSession, $iDelay = Default, $iTimeout = Default, $sElement =
 	Return SetError(__WD_Error($sFuncName, $iErr, $sMessage, $iExt), $iExt, $iReturn)
 EndFunc   ;==>_WD_LoadWait
 
+; #FUNCTION# ====================================================================================================================
+; Name ..........: _WD_LocateElement
+; Description ...: Search the current document (including frames) and return locations of matching elements
+; Syntax ........: _WD_LocateElement($sSession, $sStrategy, $sSelector[, $bShadowRoot = Default])
+;                  $iErr = @error[, $iExt = @extended]]]])
+; Parameters ....: $sSession     - Session ID from _WD_CreateSession
+;                  $sStrategy    - Locator strategy. See defined constant $_WD_LOCATOR_* for allowed values
+;                  $sSelector    - $sSelector - Indicates how the WebDriver should traverse through the HTML DOM to locate the desired element(s).
+;                  $bShadowRoot  - [optional] Starting node is a shadow root? Default is False
+; Return values .: Success - array of matching frames (format like in _WD_FrameList)
+;                  Failure - "" (empty string) and sets @error to one of the following values:
+;                  - $_WD_ERROR_GeneralError
+;                  - $_WD_ERROR_Exception
+;                  - $_WD_ERROR_NoMatch
+; Author ........: mLipok
+; Modified ......:
+; Remarks .......: Returned location (path like 'null/2/0') can be used with _WD_FrameEnter before _WD_FindElement or _WD_WaitElement will be used.
+;                  In case when $_WD_ERROR_Exception is set returned location is valid, but was not able back to calling frame,
+;                  	or some frames have become inaccessible during processing
+; Related .......: _Wd_FrameList, _WD_FindElement
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func _WD_LocateElement($sSession, $sStrategy, $sSelector, $bShadowRoot = Default)
+	Local Const $sFuncName = "_WD_LocateElement"
```

Comment:
$bShadowRoot doesn't make sense here unless you are going to add a starting node parameter.

Reply:
so we should add all this additionall parameters

`$sStartNodeID = Default, $bMultiple = Default, $bShadowRoot = Default)`

to have the same functionality as:
```autoit
Func _WD_FindElement($sSession, $sStrategy, $sSelector, $sStartNodeID = Default, $bMultiple = Default, $bShadowRoot = Default)
```

Reply:
No, you either need to add `$sStartNodeID` or remove `$bShadowRoot`. I never mentioned `$bMultiple` because it isn't needed as we always want multiple matches in this scenario, correct?

Reply:
After further consideration, I think that you should simply drop the `$bShadowRoot` parameter because it doesn't make sense in this scenario where you are trying to locate matching elements across all available frames.

Reply:
Take a look on: [$bShadowRoot removed and cleanup parameters](https://github.com/Danp2/au3WebDriver/pull/433/commits/c81cdd1ee7054cb34b77cc31414f6e677e136dd4)